lms/services/vitalsource/client.py

```diff
@@ -52,9 +60,21 @@ def book_toc(self, book_id: str):
 
             raise
 
-        return BookTOCSchema(response).parse()
+        toc = BookTOCSchema(response).parse()
+        for chapter in toc["table_of_contents"]:
+            chapter["url"] = self.generate_document_url(book_id, chapter["cfi"])
```

Comment:
Here we inject the URL directly into the dict. Not sure if this is the more marshmellow-y way of doing this. 

The URL can't be generated from the data of `BookTOCSchema` itself. It needs its parent `book_id`.

Reply:
I think it's probably fine. If you want all the work to be done in the schema perhaps it can be `toc = BookTOCSchema(response).parse(book_id)`. Then the schema would have all the info it needs to inject the chapter URLs itself